documentation/modules/cruise-control/con-cruise-control-overview.adoc

```diff
@@ -2,54 +2,508 @@
---
-
-Other Cruise Control features are not currently supported, including self healing, notifications, and write-your-own goals.
+Cruise Control comprises four main components: 
+
+Load Monitor:: Load Monitor collects the metrics and analyzes cluster workload data.
+Analyzer:: Analyzer generates optimization proposals based on collected data and configured goals.
+Anomaly Detector:: Anomaly Detector identifies and reports irregularities in cluster behavior.
+Executor:: Executor applies approved optimization proposals to the cluster.
+
+Cruise Control also provides a REST API for client interactions, which Strimzi uses to support these features:
+
+* Generating optimization proposals from optimization goals
+* Rebalancing a Kafka cluster based on an optimization proposal
+* Changing topic replication factor
+
+NOTE: Other Cruise Control features are not currently supported, including self healing, notifications, and write-your-own goals.
+
+== Optimization goals
+
+Optimization goals define objectives for rebalancing, such as distributing topic replicas evenly across brokers. 
+
+They are categorized as follows:
+
+* *Supported goals* are a list of goals supported by the Cruise Control instance that can be used in its operations. 
+By default, this list includes all goals included with Cruise Control. 
+For a goal to be used in other categories, such as default or hard goals, it must first be listed in supported goals. 
+To prevent a goal’s usage, remove it from this list.
+* *Hard goals* are preset and must be satisfied for a proposal to succeed.
+* *Soft goals* are preset goals with objectives that are prioritized during optimization as much as possible, without preventing a proposal from being created if all hard goals are satisfied.
+
+* *Default goals* refer to the goals used by default when generating proposals. 
+They match the supported goals unless specifically set by the user.
+* *Proposal-specific goals* are a subset of default goals configured for specific proposals.
+
+Configure optimization goals in the `Kafka` and `KafkaRebalance` custom resources.
+
+* `Kafka` resource for supported, hard, and default goals.
+** Supported goals: `Kafka.spec.cruiseControl.config.goals`
+** Hard goals: `Kafka.spec.cruiseControl.config.hard.goals`
+** Default goals: `Kafka.spec.cruiseControl.config.default.goals`
+* `KafkaRebalance` resource for proposal-specific goals.
+** Proposal-specific goals: `KafkaRebalance.spec.goals`
+
+=== Supported goals
+
+Supported goals are predefined and available to use for generating Cruise Control optimization proposals. 
+Goals not listed as supported goals cannot be used in Cruise Control operations.
+Some supported goals are preset as hard goals.
+
+To simplify configuration, use the inherited supported goals unless you need to exclude specific goals from `KafkaRebalance` resources. 
+You can adjust the priority order in the default optimization goals configuration.
```

Comment:
Since this is about the `default.goals` list, maybe we should move this line to the "Default goals" section